uxarray/grid/grid.py

```diff
@@ -85,13 +89,15 @@ def __init__(self, input_obj, **kwargs):
         # initialize attributes
         self._antimeridian_face_indices = None
 
-        # initialize cached data structures
+        # initialize cached data structures (visualization)
         self._gdf = None
         self._poly_collection = None
-        # TODO: fix when adding/exercising gridspec
 
-        # unpack kwargs
-        # sets default values for all kwargs to None
+        # initialize cached data structures (nearest neighbor operations)
+        self._corner_node_balltree = None
+        self._center_node_balltree = None
```

Comment:
I've adopted these changes in the following manner, since there is value to allowing queries on either the face centers or corner nodes in a single workflow without needing to reinitialize the `BallTree`
* The grid now has a single `Grid.ball_tree` attribute/property
* Each `BallTree` is now written to store both a corner node and face center wrapped ball tree internally
* The user can indicate which tree to use for queries by passing a `tree_type` 

```Python
# defaults to "corner nodes"
uxgrid.ball_tree.query([180.0, 0.0],k=1)

# explicitly specify "corner nodes"
uxgrid.ball_tree.query([180.0, 0.0],k=1, tree_type="corner nodes")

# explicitly specify "face centers"
uxgrid.ball_tree.query([180.0, 0.0],k=1, tree_type="face centers")